### PR TITLE
[EBPF] Allow --compile-only and cross-compile in kmt.build

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

Enables cross-compilation in the `kmt.build` task. Also adds an argument to build the binary without copying it to remote VMs.

### Motivation

Enable testing `system-probe´ in different architectures.

### CODE REVIEW

File: tasks/kmt.py [hunk 1]
```diff
@@ -1113,35 +1114,39 @@ def build(
...
+
+    build_task = "build-sysprobe-binary" if component == "system-probe" else "build"
+    cc.exec(
+        f"cd {CONTAINER_AGENT_PATH} && git config --global --add safe.directory {CONTAINER_AGENT_PATH} && inv -e {component}.{build_task} --no-bundle --arch={arch_obj.name}",
+    )
+
+    cc.exec(f"tar cf {CONTAINER_AGENT_PATH}/kmt-deps/{stack}/build-embedded-dir.tar {EMBEDDED_SHARE_DIR}")
+
+    if compile_only:
+        return
```
Comment (val06) [inline tasks/kmt.py:1133]: consider (if not too much work) extracting the remaining of this function to another function and call it only if compile_only == False

**Review by val06 — APPROVED** [review]
approved with suggestion

**Comment by brycekahle** [comment]
> Also adds an argument to build the binary without copying it to remote VMs.

What is the use case of this?

**Comment by gjulianm** [comment]
> > Also adds an argument to build the binary without copying it to remote VMs.
> 
> What is the use case of this?

Mostly for testing and development of KMT, and parity with `kmt.prepare` which has a `--compile-only` flag too.

**Pull request merged**